### DISCUSSION TITLE
Error caused by having composer/composer as a dev dependency. Is this normal?

### DISCUSSION THREAD

**Discussion by rhysemmerson** [composer/composer#10502] [Q&A / Support]
We have a project that started failing to deploy and I've pinned it down to the fact that the project has `composer/composer` as a dev dependency (not directly but via other packages).

```
> composer why composer/composer
barryvdh/laravel-ide-helper  v2.12.1  requires  composer/composer (^1.10.23 || ^2.1.9)  
nunomaduro/phpinsights       v2.1.0   requires  composer/composer (^2.0) 
```

When the pipeline runs the tests it first runs `composer install` which installs `composer/composer` in the vendor directory and links the binary in `vendor/bin`.

The deploy step then runs `composer install --no-dev` but this time instead of using the global binary it uses the one in `vendor/bin`, this is then removed by itself since it's a dev dependency. Then when it tries to autoload a file from within `composer/composer` we get a file not found error.

Is it normal to have `composer/composer` as a dependency?

**Comment by Seldaek** [comment]
For the no-dev install I think you should definitely use the global one, that should help.. but you may still encounter some issues if you require composer/composer in the project. It's really not so recommended.
